pyzeebe/job/job.py

```diff
@@ -58,17 +58,19 @@ def set_failure_status(self, message: str) -> None:
 
         """
         if self.zeebe_adapter:
-            self.zeebe_adapter.fail_job(job_key=self.key, message=message)
+            self.zeebe_adapter.fail_job(job_key=self.key, retries=self.retries, message=message)
```

Comment:
We should decrement one from retries here. From Zeebe's documentation:

> Marks the job as failed; if the retries argument is positive, then the job will be immediately activatable again, and a worker could try again to process it. If it is zero or negative however, an incident will be raised, tagged with the given errorMessage, and the job will not be activatable until the incident is resolved.

So the job would be retried endlessly unless we do.

Reply:
You're absolutely right, I though naively that the retries would be decremented on Zeebe side, after testing that's not the case. Changing it